### PR TITLE
ddl: add global-state-syncer for handling data to etcd. 

### PR DESCRIPTION
**Pull request by zimulala**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/43112

Problem Summary:

### What is changed and how it works?
Add some interfaces for stateSyncer as following:
```
type StateSyncer interface {
	Init(ctx context.Context) error
	UpdateGlobalState(ctx context.Context, stateInfo *StateInfo) error
	GetGlobalState(ctx context.Context) (*StateInfo, error)
	IsUpgradingState() bool
	WatchChan() clientv3.WatchChan
	Rewatch(ctx context.Context)
}
```

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: ddl/syncer/state_syncer_test.go [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+	"github.com/pingcap/tidb/store/mockstore"
+	"github.com/pingcap/tidb/util"
+	"github.com/stretchr/testify/require"
+	"go.etcd.io/etcd/tests/v3/integration"
+)
+
+func TestStateSyncerSimple(t *testing.T) {
+	variable.EnableMDL.Store(false)
+	if runtime.GOOS == "windows" {
+		t.Skip("integration.NewClusterV3 will create file contains a colon which is not allowed on Windows")
```
Comment (Defined2014) [inline ddl/syncer/state_syncer_test.go:37]: Do we now support compiling TiDB on windows?

File: ddl/syncer/state_syncer.go [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+	if len(kvs) != 1 {
+		if len(kvs) > 0 {
+			return nil, errors.Errorf("get key value count:%d wrong", len(kvs))
+		}
+		s.clusterState = state
+		return state, nil
+	}
+	err = state.Unmarshal(kvs[0].Value)
+	if err != nil {
+		logutil.BgLogger().Info("get global state failed", zap.String("key", s.etcdPath), zap.ByteString("value", kvs[0].Value), zap.Error(err))
```
Comment (Defined2014) [inline ddl/syncer/state_syncer.go:180]: why log level is info here?

File: ddl/syncer/state_syncer_test.go (hunk 1 above)
Comment (zimulala) [reply ddl/syncer/state_syncer_test.go:37]: I'm not sure. I think that's supportive. I checked, and the `SecureBootstrap` is not supported.

**Comment by zimulala** [comment]
PTAL @Defined2014

File: ddl/syncer/state_syncer_test.go (hunk 1 above)
Comment (Defined2014) [reply ddl/syncer/state_syncer_test.go:37]: I mean maybe wo could just let it compile failed. But it's not a serious issue. :)

File: ddl/syncer/state_syncer.go [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+	IsUpgradingState() bool
+	WatchChan() clientv3.WatchChan
+	Rewatch(ctx context.Context)
+}
+
+// StateInfo is the tidb cluster state.
+// It will not be updated when the tidb cluster upgrading.
+type StateInfo struct {
+	State   string `json:"state"`
+	Version string `json:"version"`
```
Comment (Defined2014) [inline ddl/syncer/state_syncer.go:63]: What is the role of version here? For tidb version or something else? If we needn't it in this PR, maybe add it later?

Comment (zimulala) [reply ddl/syncer/state_syncer.go]: Done.

**Review by Defined2014 — APPROVED** [review]

**Comment by zimulala** [comment]
PTAL @tangenta

**Comment by zimulala** [comment]
Found a failed UT: https://github.com/pingcap/tidb/issues/43170

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**